icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/NumberFormatTest.java

```diff
@@ -6966,4 +6966,94 @@ public void Test21556_CurrencyAsDecimal() {
             assertEquals("Via applyPattern: field position end", 3, fp.getEndIndex());
         }
     }
+
+    @Test
+    public void TestParseWithEmptyCurr() {
+        DecimalFormat df = (DecimalFormat)NumberFormat.getInstance(new ULocale("en_US"), NumberFormat.CURRENCYSTYLE);
+        DecimalFormatSymbols dfs = df.getDecimalFormatSymbols();
+        dfs.setCurrencySymbol("");
+        df.setDecimalFormatSymbols(dfs);
+ 
+        String textToParse = "3";    
+        ParsePosition ppos = new ParsePosition(0);
+
+        Number num = df.parse(textToParse, ppos);
+        if (ppos.getIndex() != 1 || num.doubleValue() != 3.0) {
+            errln("parse with empty curr symbol, expect pos 1, value 3.0; get "
+                                + ppos.getIndex() + ", " + num.doubleValue());
+        }
+
+        ppos.setIndex(0);
+        CurrencyAmount currAmt = df.parseCurrency(textToParse, ppos);
+        if (currAmt != null) {
+            errln("parseCurrency with empty curr symbol, expect null but succeeds");
+        }
```

Comment:
This kind of threw me.  Intuitively, it seems like if the currency symbol is the empty string and you parse a string that doesn't have a currency symbol, it should succeed-- the "currency symbol" in the string matches the "currency symbol" in the formatter.  Why is this not the expected behavior?

IIRC, we see this in practice in a couple of locales that have an infix currency symbol, such as the Cape Verdean escudo.  The "currency symbol" is blank, and the actual currency symbol is the currency decimal separator.  In that case, you'd get a parse error if the currency decimal separator wasn't the expected one.  Is this related to that kind of use case?

Reply:
@richgillam parseCurrency is intended to parse an arbitrary currency, not the specific currency of the locale associated with the formatter. So if there is no currency symbol itcnnot report which currency it parsed. The standard parse is only intended to parse a specific currency, either the one for the locale or some other symbol that has been specifically set (which might be empty).

Reply:
Oh, okay.  That makes total sense then.  Thanks for the explanation.